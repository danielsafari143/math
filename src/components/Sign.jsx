import PropTypes from 'prop-types';

function Sign({ symbole, color, func }) {
  return <input onClick={(e) => func(e)} className={`signs ${color}`} type="button" value={symbole} />;
}

Sign.propTypes = {
  symbole: PropTypes.node.isRequired,
  color: PropTypes.node.isRequired,
  func: PropTypes.node.isRequired,
};

export default Sign;
