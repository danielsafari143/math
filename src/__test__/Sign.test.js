import renderer from 'react-test-renderer';
import Sign from '../components/Sign';

it('Test if the UI matches ', () => {
  const handleEquale = jest.fn((e) => e.target.value);
  const tree = renderer.create(<Sign symbole={2} color="gray" func={handleEquale} />).toJSON();
  expect(tree).toMatchSnapshot();
});
