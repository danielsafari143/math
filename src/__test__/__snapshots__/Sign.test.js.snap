// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the UI matches  1`] = `
<input
  className="signs gray"
  onClick={[Function]}
  type="button"
  value={2}
/>
`;
